### PR TITLE
Form validation branch

### PR DESCRIPTION
**Pull request by Metsanpeitto**
- Process user input according to business rules.
- Use client-side validation to catch and throw errors in the UI.

### CODE REVIEW

File: js/input_validation.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ *   This is the Input Validation function.
+ *   It checks that only lowercase is being typed into
+ *   the email input of the contact form.
+ */
```
Comment (vichuge) [inline js/input_validation.js:5]: - [ ] Please, delete the commented lines, this could help to have a clean code and easy to read. 🐶

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Good points 👍 
- No Linters error ✔️

## Aspects to improve ♻️

- [ ] It looks like the text appears next to the input, I kindly invite you to show the text closer to the submit button (at top of it, for example):
image

- [ ] Check the comments under the review.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/input_validation.js (hunk 1 above)
Comment (Metsanpeitto) [reply js/input_validation.js:5]: Thanks for pointing out the mistaken location where the form_validation_error message shows. I have already relocated to the place you said.

About deleting the comments, sorry, and please excuse me that I contradict you. Those comments have a function and are needed to help to understand the code. They are part of the best practices and their use should be encouraged, not the opposite. If there is a better way of using them, please let me know. I do the best I can but I am open to improvement. 

I quote :
"Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. Remember, well-documented code is as important as correctly working code."

From: [https://www.cs.utah.edu/~germain/PPS/Topics/commenting.html#:~:text=Commenting%20involves%20placing%20Human%20Readable,that%20other%20people%20will%20use.](URL)

**Comment by Davidosky007** [comment]
Hi @Metsanpeitto, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**